lang/values.md

```diff
@@ -103,6 +103,7 @@ impl Type {
             throw!();
         }
         // Fails if any byte is `Uninit`.
+        // FIXME: We want a List::try_map supporting options.
```

Comment:
so, currently `List::try_map` only works for `NdResult` (same for `Option::try_map`).

I originally implemented a general `Option::try_map` function, but iirc it wasn't able to infer types, and was thus a bit unpractical.

@RalfJung I presume you'd prefer a general `.try_map` impl.
If you want to, I can revisit my original impl

Reply:
Yeah would be nice if it could be as general as `try_collect`.

Reply:
Worked out using the `Residual` trait